packages/airnode-protocol/contracts/rrp/AirnodeRrpV0DryRun.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+/// @title Contract that complements Airnode request–response protocol (RRP) to
+/// allow Airnode to estimate the gas required to execute a fulfillment
+/// @dev Typically, contracts are built to revert when an external call they
+/// make reverts. In contrast, AirnodeRrpV0 does not revert when the external
+/// call during the fulfillment reverts, and instead fails gracefully by
+/// emitting a `FailedRequest` event. This event signals to the future
+/// invocations of the stateless Airnode to not retry the failed fulfillment.
+/// Although this approach meets the intended purpose, it disables Airnode from
+/// calling `eth_estimateGas` on `fulfill()` to estimate the gas amount that
+/// will be used to execute a fulfillment successfully. Specifically, since
+/// `eth_estimateGas` looks for the lowest gas limit that results in the
+/// transaction not reverting, and AirnodeRrpV0's `fulfill()` does not revert
+/// when its external call reverts (because it runs out of gas),
```

Comment:
are you saying it will always return the gas estimate for when the airnode emits `Failed` ?

Reply:
Yes. This was originally picked up by e2e tests for https://github.com/api3dao/airnode/issues/1589 was failing.